### PR TITLE
fix: fix logger src file

### LINKED ISSUE #641: Logger is not pointing to source file
**Description**

Logger is not pointing to source file.
Logger always points to utils file where log function is.

**To Reproduce**
1. Change log LEVEL
2. Load a page with a sidekick
3. Check console

**Expected behavior**

Logger should pointing to source file where log function is used.

**Screenshots**

Screenshot 2023-12-28 at 12 48 07

### PR DESCRIPTION
**Pull request by easingthemes**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Fix #641

### CODE REVIEW

**Comment by easingthemes** [comment]
Current:
Screenshot 2023-12-28 at 12 48 07
After PR:
Screenshot 2023-12-28 at 12 46 52

File: src/extension/utils.js [hunk 1]
```diff
@@ -18,15 +18,27 @@ export const DEV_URL = 'http://localhost:3000';
...
+    info: { level: 2, color: '' },
+    warn: { level: 1, color: 'orange' },
+    error: { level: 0, color: 'red' },
+  };
+  Object.keys(logLevels).forEach((method) => {
+    const color = `color: ${logLevels[method].color}`;
+    const prefix = `%c[${method.toUpperCase()}]`;
+    /* eslint-disable no-console */
+    log[method] = log.LEVEL > logLevels[method].level && console.log.bind(console, prefix, color);
+    /* eslint-enable no-console */
```
Comment (rofe) [inline src/extension/utils.js:36]: The methods of higher log levels still need to be defined for the rest of the code to work:
```suggestion
    log[method] = log.LEVEL > logLevels[method].level
      // eslint-disable-next-line no-console
      ? console[method].bind(console, prefix, color)
      : () => {};
```

**Review by rofe — CHANGES_REQUESTED** [review]
Nice! However, I found 2 problems with this approach:
1. Calls to `log` methods outside of the current log level fail
2. The [current test method](https://github.com/adobe/helix-sidekick-extension/blob/main/test/extension/utils.test.js#L89-L95) no longer works, possibly due to the binding

File: src/extension/utils.js (hunk 1 above)
Comment (easingthemes) [reply src/extension/utils.js]: Updated.

File: src/extension/utils.js [hunk 2]
```diff
@@ -18,15 +18,41 @@ export const DEV_URL = 'http://localhost:3000';
...
+
+  createMethods() {
+    Object.keys(this.config).forEach((method) => {
+      const color = `color: ${this.config[method].color}`;
+      const prefix = `%c[${method.toUpperCase()}]`;
+      this[method] = this.LEVEL > this.config[method].level
+        // eslint-disable-next-line no-console
+        ? console[method].bind(console, prefix, color)
+        : () => {};
+      /* eslint-enable no-console */
```
Comment (rofe) [inline src/extension/utils.js:50]: Not necessary
```suggestion
```

**Review by rofe — APPROVED** [review]
LGTM - apart from the now obsolete `eslint` instruction

File: src/extension/utils.js (hunk 2 above)
Comment (easingthemes) [reply src/extension/utils.js]: Removed

**Pull request merged**